packages/language-server/test/ui5-model-spec.ts

```diff
@@ -184,10 +213,278 @@ describe("the UI5 language assistant ui5 model", () => {
             };
           },
           cachePath,
-          ""
+          undefined,
+          undefined,
+          true
         );
         expect(fetcherCalled).to.be.true;
       }).timeout(GET_MODEL_TIMEOUT);
     });
   });
+
+  describe("version negotiation", async () => {
+    let cachePath: string;
+    let cleanup: () => Promise<void>;
+    const versionMap = {
+      latest: {
+        version: "1.105.0",
+        support: "Maintenance",
+        lts: true,
+      },
+      "1.105": {
+        version: "1.105.0",
+        support: "Maintenance",
+        lts: true,
+      },
+      "1.96": {
+        version: "1.96.11",
+        support: "Maintenance",
+        lts: true,
+      },
+      "1.84": {
+        version: "1.84.27",
+        support: "Maintenance",
+        lts: true,
+      },
+      "1.71": {
+        version: "1.71.50",
+        support: "Maintenance",
+        lts: true,
+      },
+    };
+    const versionInfo = {
+      libraries: [
+        {
+          name: "sap.ui.core",
+        },
+      ],
+    };
+    const createResponse = (ok: boolean, status: number, json?: unknown) => {
+      return {
+        ok,
+        status,
+        json: async (): Promise<unknown> => {
+          return json;
+        },
+      };
+    };
+
+    beforeEach(async () => {
+      ({ path: cachePath, cleanup } = await tempFile());
+    });
+
+    afterEach(async () => {
+      await cleanup();
+    });
+
+    it("resolve the default version", async () => {
+      expect(
+        await negotiateVersionWithFetcher(
+          async (url: string): Promise<FetchResponse> => {
+            if (url.endsWith("/version.json")) {
+              // request for version mapping info
+              return createResponse(true, 200, versionMap);
+            } else {
+              // request for version info (needs to be just an object => found!)
+              return createResponse(true, 200, versionInfo);
+            }
+          },
+          cachePath,
+          FRAMEWORK,
+          VERSION
+        )
+      ).to.be.equal(VERSION);
+    });
+
+    it("resolve available concrete version", async () => {
+      expect(
+        await negotiateVersionWithFetcher(
+          async (url: string): Promise<FetchResponse> => {
+            if (url.endsWith("/version.json")) {
+              // request for version mapping info
+              return createResponse(true, 200, versionMap);
+            } else {
+              // request for version info (needs to be just an object => found!)
+              return createResponse(true, 200, versionInfo);
+            }
+          },
+          cachePath,
+          FRAMEWORK,
+          "1.105.0"
+        )
+      ).to.be.equal("1.105.0");
+      expect(
```

Comment:
should not this be two separate tests? one for 105 and one for 104?

Reply:
I don't mind on this

Reply:
Splitted